client/state/notices/middleware.js

```diff
@@ -141,6 +146,36 @@ export const onPublicizeConnectionUpdateFailure = ( dispatch, { error } ) => dis
 	} ) )
 );
 
+export const onThemeUploadSuccess = ( dispatch, { themeId } ) => {
+	return dispatch( successNotice(
+		translate( 'Successfully uploaded theme %(name)s', {
+			args: {
+				name: themeId
```

Comment:
Actually, we can pass the `theme` object [here](https://github.com/Automattic/wp-calypso/blob/c9474af93c4809c313f1a65025d636a998badf89/client/state/themes/actions.js#L470-L477)! 🎉